lib/bolt/bolt_option_parser.rb

```diff
@@ -863,7 +863,7 @@ def initialize(options)
       end
       define('--log-level LEVEL',
              "Set the log level for the console. Available options are",
-             "debug, info, notice, warn, error, fatal, any.") do |level|
+             "trace, debug, info, warn, error, fatal, any.") do |level|
```

Comment:
It seems like we should leave `notice` here since it can be configured in Bolt config files, and that's what sets the console config for messages from Puppet. For example, this will break plans that call `notice()` in their apply blocks.

Reply:
Or, alternatively, we should remove notice from both the CLI and config options. https://github.com/puppetlabs/bolt/blob/main/lib/bolt/config/options.rb#L189

Reply:
This was meant as a sort of soft deprecation of `notice`. No messages should be going to that level now, but if you've specified it as the level for a log in your config, we don't want that to break. But we don't want anybody to actually _set_ their log level to `notice` now, so I removed it from the documentation.

I will update the config in `options.rb` to remove `notice` and also add `trace`. I wasn't sure if that code was used for validation so I left `notice` for compatibility. But now I see it's just for docs/schema generation, so it can be safely updated.

Reply:
Should we have a deprecation warning if you try to pass `notice`? It may be annoying to silently have it go from working to not working, even on the CLI.

Reply:
Just did some more digging: It appears that `Thread#name` was added in Ruby 2.3 which is why `logging`, which _well_ predates that Ruby version, uses a `:name` variable instead. We might be able to get them to accept a PR to check the `name` attr.

I was on the fence about warning for `notice` at this point. We should definitely warn about it before we remove it, but I wanted to let it sit a bit and make sure the new levels actually work for people before committing to removing it for real.